include/envoy/router/router.h

```diff
@@ -478,6 +494,13 @@ class RouteEntry : public ResponseEntry {
    * nullptr is returned.
    */
   virtual const Protobuf::Message* perFilterConfig(const std::string& name) const PURE;
+
+  /**
+   * @return const PerRouteConfigObject* the per-filter config pre-processed object for
+   *  the given filter name. If not per-filter config for the filter, or the filter factory
+   *  doesn't create such object, nullptr is returned.
+   */
+  virtual const PerRouteConfigObject* perFilterConfigObject(const std::string& name) const PURE;
```

Comment:
IF we are going down the path of merging with perFilterConfig, I would kill this function and change signature of the other..

Reply:
+1